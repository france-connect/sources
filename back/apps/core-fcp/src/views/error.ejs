<!doctype html>
<html lang="fr" data-fr-scheme="light">
<head>
  <%- include('includes/head.ejs') %>
  <link rel="stylesheet" href="/css/pages/error.css">
  <%- include('pages/error/error.page-title.ejs') %>
</head>
<body id="fc-error-page">
  <div class="fc-sticky-wrapper">
    <div class="fc-sticky-body">
      <%- include('includes/header.ejs') %>
      <main class="fr-container fr-mt-5w fr-mt-md-8w">
        <div class="fr-grid-row fr-grid-row--gutters">
          <section
            data-testid="error-section"
            class="fr-col-12 fr-col-offset-md-1 fr-col-md-9 fr-mb-6v fr-mb-md-10v">
            <%- include('pages/error/error.welcome.ejs') %>
            <%- include('pages/error/error.action.ejs') %>
          </section>
        </div>

        <div class="fr-grid-row fr-grid-row--gutters fr-grid-row--middle">
          <%- include('pages/error/error.support.ejs') %>
        </div>

      </main>
    </div>
    <%- include('includes/footer.ejs') %>
  </div>
  <%- include('includes/scripts.ejs') %>
</body>
</html>
