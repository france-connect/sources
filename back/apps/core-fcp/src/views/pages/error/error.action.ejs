<% if (locals.message) { %>
<h2 class="fr-h6 fr-mb-2w" data-testid="action">
  Que faire ?
</h2>
<p class="fr-text--md fr-mb-0" data-testid="error-message">
  <%= locals.message %>
</p>
<% } %>
