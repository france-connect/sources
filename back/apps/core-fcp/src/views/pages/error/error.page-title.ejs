<title>Connexion - Erreur - <%= locals.config?.App?.platform %></title>
