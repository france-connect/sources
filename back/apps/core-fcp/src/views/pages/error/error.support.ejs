<%
const showSupportButton = locals.config?.Core?.supportFormCodes?.includes(locals.code);
const offsetLgClass = showSupportButton
  ? 'fr-col-offset-lg-0'
  : 'fr-col-offset-lg-1';
%>

<% if (showSupportButton) { %>
<%
const { OidcClient } = locals.session;
const sp_name = OidcClient?.spName || locals.spName;
const idp_name = OidcClient?.idpLabel || OidcClient?.idpName;

const params = {
  platform: locals.config.App?.name,
  error_code: locals.code,
  sp_name,
  idp_name,
  error_id: locals.id,
};

const queryString = Object.entries(params).reduce((acc, [key, value]) => {
  const separator = acc === '' ? '?' : '&';
  return value ? `${acc}${separator}${key}=${encodeURIComponent(value)}` : acc;
}, '');

const href = `${locals.config.Core.supportFormUrl}${queryString}`;
%>
<section
  data-testid="error-support"
  class="fr-col-12 fr-col-offset-md-1 fr-col-md-10 fr-col-offset-lg-1 fr-col-lg-3 fr-mb-2w">
  <a
    href="<%= href %>"
    rel="noreferrer noopener"
    data-testid="form-link-container"
    class="fr-btn fr-btn--lg fc-btn--full-width fc-text--center"
    title="Naviguer vers le formulaire dâ€™aide"
  >
    Contactez-nous
  </a>
</section>
<% } %>

<section
  data-testid="error-back"
  class="fr-col-12 fr-col-offset-md-1 fr-col-md-10 <%= offsetLgClass %> fr-col-lg-8 fr-mb-2w fc-text--center fc-text-lg--left">
  <%- include('includes/back-button.ejs') %>
</section>
