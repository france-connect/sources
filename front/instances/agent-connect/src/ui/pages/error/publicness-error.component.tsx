import classnames from 'classnames';
import React from 'react';
import { useMediaQuery } from 'react-responsive';

import styles from './error.module.scss';

export const PublicnessErrorComponent = React.memo(() => {
  const gtTablet = useMediaQuery({ query: '(min-width: 768px)' });

  return (
    <div
      className={classnames(styles.page, 'fr-m-auto', {
        // Class CSS
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-mt-4w': !gtTablet,
        // Class CSS
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-mt-8w': gtTablet,
      })}>
      <h2
        className={classnames(styles.title, 'fr-text--bold fr-px-2w', {
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'text-center': gtTablet,
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'text-left': !gtTablet,
        })}>
        Accès impossible
      </h2>
      <div className="fr-mx-auto fr-my-4w fr-px-2w fr-text--lg">
        <p className="fr-mb-4w">Ce service est réservé aux agents de la Fonction Publique.</p>
        <p className="fr-mb-4w">
          Vous êtes rattaché à une organisation de droit privé. Vous ne pouvez pas y avoir accès.
        </p>
      </div>
    </div>
  );
});

PublicnessErrorComponent.displayName = 'ErrorComponent';
