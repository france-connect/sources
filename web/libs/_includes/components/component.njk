{% macro component(name, params = {}) %}
  {% include "components/" + name + ".njk" ignore missing %}
{% endmacro %}