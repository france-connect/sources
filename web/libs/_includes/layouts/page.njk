---
layout: layouts/base.njk
---
<div class="fr-container fr-py-8v">
  {% if showBreadcrumb %}
    {% include "components/breadcrumb.njk" %}
  {% endif %}
  <div class="fr-grid-row" data-pagefind-body>
    <div class="fr-col-12">
      {{ content | safe }}
    </div>
  </div>
</div>