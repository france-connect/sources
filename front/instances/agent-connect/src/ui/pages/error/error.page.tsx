import React from 'react';
import { Helmet } from 'react-helmet';

import { ErrorComponent } from './error.component';
import { NotFoundComponent } from './not-found.component';
import { PublicnessErrorComponent } from './publicness-error.component';

const PageTitle = () => (
  <Helmet>
    <title>AgentConnect - Erreur</title>
  </Helmet>
);

// a temporary way to display a specific error content
// because react app is supposed to be replace by ejs pages soon
// if fca react app is not deleted at the end of 2023,
// this part should be rewrite
export const AGENT_NOT_FOUND_ERROR_CODE = 'Y000015';

export const ErrorPage = React.memo(() => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const hasError = (window as any).appError !== undefined;
  if (hasError) {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    if ((window as any).appError.code === AGENT_NOT_FOUND_ERROR_CODE) {
      return (
        <React.Fragment>
          <PageTitle />
          <PublicnessErrorComponent />
        </React.Fragment>
      );
    }

    return (
      <React.Fragment>
        <PageTitle />
        <ErrorComponent
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          errors={(window as any).appError}
        />
      </React.Fragment>
    );
  }
  return (
    <React.Fragment>
      <PageTitle />
      <NotFoundComponent />
    </React.Fragment>
  );
});

ErrorPage.displayName = 'ErrorPage';
