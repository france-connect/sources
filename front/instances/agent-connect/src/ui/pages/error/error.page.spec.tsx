import { render } from '@testing-library/react';
import { useMediaQuery } from 'react-responsive';

import { ErrorComponent } from './error.component';
import { AGENT_NOT_FOUND_ERROR_CODE, ErrorPage } from './error.page';
import { NotFoundComponent } from './not-found.component';
import { PublicnessErrorComponent } from './publicness-error.component';

jest.mock('./error.component');
jest.mock('./not-found.component');
jest.mock('./publicness-error.component');

describe('ErrorPage', () => {
  it('should match the snapshot, in a desktop viewport', () => {
    // given
    jest.mocked(useMediaQuery).mockReturnValueOnce(true);

    // when
    const { container } = render(<ErrorPage />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, in a mobile viewport', () => {
    // given
    jest.mocked(useMediaQuery).mockReturnValueOnce(true);

    // when
    const { container } = render(<ErrorPage />);

    // then
    expect(container).toMatchSnapshot();
  });

  it('should render the NotFoundComponent when window.appError is undefined', () => {
    // when
    render(<ErrorPage />);

    // then
    expect(NotFoundComponent).toHaveBeenCalledTimes(1);
  });

  it('should render the ErrorComponent when window.appError is defined', () => {
    // given
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (window as any).appError = 'something';

    // when
    render(<ErrorPage />);

    // then
    expect(ErrorComponent).toHaveBeenCalledTimes(1);
    expect(ErrorComponent).toHaveBeenCalledWith({ errors: 'something' }, {});
  });

  it('should render the PublicnessErrorComponent when window.appError.code is AGENT_NOT_FOUND_ERROR_CODE', () => {
    // given
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (window as any).appError = { code: AGENT_NOT_FOUND_ERROR_CODE };

    // when
    render(<ErrorPage />);

    // then
    // then
    expect(PublicnessErrorComponent).toHaveBeenCalledTimes(1);
  });
});
