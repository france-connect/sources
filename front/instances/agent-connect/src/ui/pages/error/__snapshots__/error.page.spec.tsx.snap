// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should match the snapshot, in a desktop viewport 1`] = `
<div>
  <div>
    <div>
      React Helmet
    </div>
  </div>
  <div>
    NotFoundComponent
  </div>
</div>
`;

exports[`ErrorPage should match the snapshot, in a mobile viewport 1`] = `
<div>
  <div>
    <div>
      React Helmet
    </div>
  </div>
  <div>
    NotFoundComponent
  </div>
</div>
`;
